src/DynamoDbModel.php

```diff
@@ -86,11 +86,33 @@ protected function setupDynamoDb()
 
     public function save(array $options = [])
     {
-        if (!$this->getKey()) {
-            $this->fireModelEvent('creating');
+        $create = !$this->exists;
+        if ($create && $this->fireModelEvent('creating')  === false) {
+            return false;
         }
+        if (!$create && $this->fireModelEvent('updating') === false) {
+            return false;
+        }
+
+        if ($this->fireModelEvent('saving') === false) {
+            return false;
+        }
+
+        $saved = $this->newQuery()->save();
 
-        return $this->newQuery()->save();
+        if (!$saved) {
+            return $saved;
+        }
+
+        $this->exists = true;
+        if ($create) {
+            $this->wasRecentlyCreated = true;
+            $this->fireModelEvent('created');
+        } else {
+            $this->fireModelEvent('updated');
+        }
```

Comment:
not sure how you would combine it?

We either fire the "created" or the "updated" event, but not both at the same time.  We also only set "wasRecentlyCreated" when it's been created, not when it's been updated.

Laravel's Eloquent actually splits the "save" up into two other calls, ->performInsert(...) and ->performUpdate(...) which sort of removes the excessive conditionals from "save" and moves the event firing into those methods.  Speaking of which, the "saving" event is being fired too late ... it should be fired before the creating/updating so the code can cancel the "save" before either creating/updating are attempting - fixing that now.